GraphBulkImporter/Extensions.cs

```diff
@@ -0,0 +1,103 @@
+        /// <param name="partitionKeyProperty">Which property should be used for partitionKey of the GremlinVertex, defaults to "partitionKey"</param>
+        /// <param name="label">What value should be used for the label property, defaults to the Type name of the object</param>
+        /// <returns>
+        /// A new instance of a GremlinVertex
+        /// with its id and partitionKey property values set to and <typeparamref name="idProperty"/> and <typeparamref name="partitionKeyProeprty"/> properties respectively
+        /// and the label property set to the value of <typeparamref name="label"/>
+        /// </returns>
+        public static GremlinVertex ToGremlinVertex(this object obj, string idProperty,  string partitionKeyProperty, string label)
+        {
+            if (string.IsNullOrEmpty(idProperty))
+            {
+                throw new ArgumentException($"{nameof(idProperty)} cannot be Null or Empty", nameof(idProperty));
+            }
+
+            if (string.IsNullOrWhiteSpace(partitionKeyProperty))
+            {
+                throw new ArgumentException($"{nameof(partitionKeyProperty)} cannot be Null or Empty", nameof(partitionKeyProperty));
+            }
+
+            if (string.IsNullOrWhiteSpace(label))
+            {
+                throw new ArgumentException($"{nameof(label)} cannot be Null or Empty", nameof(label));
+            }
+
+            var gv = new GremlinVertex(obj.GetPropertyValue(idProperty).ToString(), label);
+            gv.AddProperty(new GremlinVertexProperty("partitionKey", obj.GetPropertyValue(partitionKeyProperty)));
+
+            foreach (var prop in obj.GetType().GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
+            {
+                var name = prop.Name;
+                var val = prop.GetValue(obj);
+
+                if (!name.Equals("id", StringComparison.InvariantCultureIgnoreCase) && !name.Equals("partitionKey", StringComparison.InvariantCultureIgnoreCase))
+                    gv.AddProperty(new GremlinVertexProperty(name, val));
+            }
+
+            return gv;
+        }       
+        private static bool HasProperty(this object obj, string propertyName)
+        {
+            return GetPropertyInfo(obj, propertyName) != null;
+        }
+        private static PropertyInfo GetPropertyInfo(object obj, string propertyName)
+        {
+            //do a case-insensitive search for the propertyName supplied
+            //i.e. match on Id, ID, and on id
+            return obj.GetType().GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+        }
+        private static object GetPropertyValue(this object obj, string propertyName)
+        {
+            var prop = GetPropertyInfo(obj, propertyName);
+            if (prop == null) throw new ArgumentException($"No {propertyName} property found.", propertyName);
```

Comment:
can combine this with line 97 with `??`. Otherwise I always like to be consistent with my if statements and even wrap one-lines in braces but thats up to you